File: fashion_outfit_suggestions_app/lib/core/models/friend_response.dart
class FriendResponse {
  final String id;
  final String friendId;
  final String fullName;
  final String username;
  final String? avatarUrl;
  final String? status;

  FriendResponse({
    required this.id,
    required this.friendId,
    required this.fullName,
    required this.username,
    this.avatarUrl,
    this.status,
  });

  factory FriendResponse.fromJson(Map<String, dynamic> json) {
    return FriendResponse(
      id: json['id']?.toString() ?? '',
      friendId: json['friendId']?.toString() ?? '',
      fullName: json['fullName'] as String? ?? '',
      username: json['username'] as String? ?? '',
      avatarUrl: json['avatarUrl'] as String?,
      status: json['status'] as String?,
    );
  }

  FriendResponse copyWith({
    String? id,
    String? friendId,
    String? fullName,
    String? username,
    String? avatarUrl,
    String? status,
  }) {
    return FriendResponse(
      id: id ?? this.id,
      friendId: friendId ?? this.friendId,
      fullName: fullName ?? this.fullName,
      username: username ?? this.username,
      avatarUrl: avatarUrl ?? this.avatarUrl,
      status: status ?? this.status,
    );
  }
}

File: backend/src/main/java/com/example/social/dto/response/FriendResponse.java
package com.example.social.dto.response;

import com.example.social.entity.enums.FriendshipStatus;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.UUID;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class FriendResponse {
    private UUID id;
    private UUID friendId;
    private String fullName;
    private String username;
    private String avatarUrl;
    private FriendshipStatus status;
}
